### ISSUE TITLE
Reindent incorrect when braces are omitted

### ISSUE THREAD

**Issue by dmurdoch** [rstudio/rstudio#7904]
### System details

    RStudio Edition : Desktop or Server 
    RStudio Version : 1.3.1093
    OS Version      :  MacOS 10.15.6
    R Version       :   4.0.2

### Steps to reproduce the problem

Enter this code in the code editor:

```r
badfn <- function() {
  obj <- 1
  for (attr in letters)
    if (attr == "z") 
      obj <- 2
  obj
}
```

and then highlight all and hit Cmd-I (or Ctrl-I) for Reindent Lines.  The last line ends up indented too far:

```r
badfn <- function() {
  obj <- 1
  for (attr in letters)
    if (attr == "z") 
      obj <- 2
    obj
}
```

### Describe the problem in detail

The auto-indenter seems to think that the last line is within the scope of the `for` loop, when it is really top level.  Things are okay if braces are used in the `for` loop or the `if` statement, or both.

### Describe the behavior you expected

The original source was indented properly and should not have been changed.

- [x ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [] If I am reporting a RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
Thanks for the bug report! It looks like the assignment within the `if` body confuses the auto-indenter. For example, this is fine (with `|` indicating cursor position):

```
for (x in 1)
  if (y)
    foo
|
```

But this is not:

```
for (x in 1)
  if (y)
    foo <- 1
  |

**Comment by kfeinauer** [comment]
Verified fixed 1.4.1035
